torch/csrc/jit/autodiff.h

```diff
@@ -12,4 +12,21 @@ namespace torch { namespace jit {
 // (aka backward) of inputs to the first stage w.r.t. the outputs of the first stage.
 void differentiate(std::shared_ptr<Graph>& graph);
 
+using value_list = std::vector<Value*>;
+struct LiftedReverse {
```

Comment:
Please make this match the Gradient structure in the pseudo code.

Reply:
This is not `Gradient` - if you look at the pseudo code, you'll see that half of its fields come from `differentiate`, while the other half come from `lambdaLiftReverse`. This lets us encapsulate the AD code in a fairly independent way, that doesn't need to know anything about `GraphExecutor`s.